src/main/java/com/igot/cb/org/service/impl/OrgServiceImpl.java

```diff
@@ -184,4 +192,189 @@ public boolean isSpvRequest(String userId, List<String> requiredRoles) {
+            log.error("Unexpected error occurred in processFrameworkCreate", e);
+        }
+        return fwName;
+    }
+
+    public static Map<String, Object> createFrameworkRequest(String channelId, String frameworkName) {
+        Map<String, Object> framework = createFramework(channelId, frameworkName);
+        Map<String, Object> request = new HashMap<>();
+        request.put("framework", framework);
+        return request;
+    }
+
+    private static Map<String, Object> createFramework(String channelId, String frameworkName) {
+        Map<String, Object> framework = new HashMap<>();
+        StringBuilder name = new StringBuilder(channelId).append("_").append(frameworkName);
+        framework.put(Constants.NAME, name);
+        framework.put(Constants.DESCRIPTION, "Master Framework Copy");
+        framework.put(Constants.CODE, name);
+        framework.put(Constants.OWNER, channelId);
+        return framework;
+    }
+
+    private void updateStatusToFailed(String orgId) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(Constants.ID, orgId);
+        map.put(Constants.FRAMEWORK_STATUS, Constants.FAILED);
+        Map<String, Object> updateOrgDetails = cassandraOperation.updateRecord(
+                Constants.KEYSPACE_SUNBIRD, Constants.ORG_TABLE, map);
+    }
+
+
+    private String frameworkReadV1(String frameworkId) {
+        String code = null;
+        try {
+            StringBuilder strUrl = new StringBuilder(cbServerProperties.getKnowledgeMS());
+            strUrl.append(cbServerProperties.getOdcsFrameworkRead()).append("/").append(frameworkId);
+            log.info("prinitng framework read url "+strUrl.toString());
+            Map<String, Object> framworkResponse = (Map<String, Object>) outboundRequestHandlerServiceImpl.fetchResult(strUrl.toString());
+            if (null != framworkResponse) {
+                if (Constants.OK.equalsIgnoreCase((String) framworkResponse.get(Constants.RESPONSE_CODE))) {
+                    Map<String, Object> resultMap = (Map<String, Object>) framworkResponse.get(Constants.RESULT);
+                    Map<String, Object> framework = (Map<String, Object>) resultMap.get(Constants.FRAMEWORK);
+                    List categoriesList = (List<Map<String,Object>>) framework.get("categories");
+                    Map<String,Object> map = (Map<String,Object>) categoriesList.get(0);
+                    code = (String) map.get(Constants.CODE);
+                    return code;
+                } else {
+                    log.info("Data not found with id : " + frameworkId);
+                }
+            } else {
+                log.info("Data not found with ID: {}", frameworkId);
+            }
+        } catch (Exception e) {
+            log.error("Failed to read framework with ID: {}", frameworkId, e);
+        }
+        return code;
+    }
+
+    private void createOrgTerm(String termName, String copyfw, String masterFramework, String orgId) {
```

Comment:
can we add createdBy for the term object ?
so that it can help who created this particular term